pkg/flowaggregator/flowrecord/record.go

```diff
@@ -0,0 +1,496 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flowrecord
+
+import (
+	"net"
+	"time"
+
+	ipfixentities "github.com/vmware/go-ipfix/pkg/entities"
+	ipfixentitiestesting "github.com/vmware/go-ipfix/pkg/entities/testing"
+	ipfixregistry "github.com/vmware/go-ipfix/pkg/registry"
+)
+
+type FlowRecord struct {
+	FlowStartSeconds                     time.Time
+	FlowEndSeconds                       time.Time
+	FlowEndSecondsFromSourceNode         time.Time
+	FlowEndSecondsFromDestinationNode    time.Time
+	FlowEndReason                        uint8
+	SourceIP                             string
+	DestinationIP                        string
+	SourceTransportPort                  uint16
+	DestinationTransportPort             uint16
+	ProtocolIdentifier                   uint8
+	PacketTotalCount                     uint64
+	OctetTotalCount                      uint64
+	PacketDeltaCount                     uint64
+	OctetDeltaCount                      uint64
+	ReversePacketTotalCount              uint64
+	ReverseOctetTotalCount               uint64
+	ReversePacketDeltaCount              uint64
+	ReverseOctetDeltaCount               uint64
+	SourcePodName                        string
+	SourcePodNamespace                   string
+	SourceNodeName                       string
+	DestinationPodName                   string
+	DestinationPodNamespace              string
+	DestinationNodeName                  string
+	DestinationClusterIP                 string
+	DestinationServicePort               uint16
+	DestinationServicePortName           string
+	IngressNetworkPolicyName             string
+	IngressNetworkPolicyNamespace        string
+	IngressNetworkPolicyRuleName         string
+	IngressNetworkPolicyRuleAction       uint8
+	IngressNetworkPolicyType             uint8
+	EgressNetworkPolicyName              string
+	EgressNetworkPolicyNamespace         string
+	EgressNetworkPolicyRuleName          string
+	EgressNetworkPolicyRuleAction        uint8
+	EgressNetworkPolicyType              uint8
+	TcpState                             string
+	FlowType                             uint8
+	SourcePodLabels                      string
+	DestinationPodLabels                 string
+	Throughput                           uint64
+	ReverseThroughput                    uint64
+	ThroughputFromSourceNode             uint64
+	ThroughputFromDestinationNode        uint64
+	ReverseThroughputFromSourceNode      uint64
+	ReverseThroughputFromDestinationNode uint64
+}
+
+// GetFlowRecord converts ipfixentities.Record to FlowRecord
+func (r *FlowRecord) GetFlowRecord(record ipfixentities.Record) {
```

Comment:
is there a reason why this is a method on `FlowRecord` instead of a function returning a new instance of a `FlowRecord`?